src/components/dialogs/VerifyEmailDialog.tsx

```diff
@@ -0,0 +1,247 @@
+import {ErrorMessage} from '#/view/com/util/error/ErrorMessage'
+import {atoms as a, useBreakpoints, useTheme} from '#/alf'
+import {Button, ButtonText} from '#/components/Button'
+import * as Dialog from '#/components/Dialog'
+import * as TextField from '#/components/forms/TextField'
+import {Loader} from '#/components/Loader'
+import {Text} from '#/components/Typography'
+
+export function VerifyEmailDialog({
+  control,
+}: {
+  control: Dialog.DialogControlProps
+}) {
+  const agent = useAgent()
+
+  const [didVerify, setDidVerify] = React.useState(false)
+
+  return (
+    <Dialog.Outer
+      control={control}
+      onClose={async () => {
+        if (!didVerify) {
+          return
+        }
+
+        try {
+          await agent.resumeSession(agent.session!)
+        } catch (e: unknown) {
+          logger.error(String(e))
+          return
+        }
+      }}>
+      <Dialog.Handle />
+      <Inner control={control} setDidVerify={setDidVerify} />
+    </Dialog.Outer>
+  )
+}
+
+export function Inner({
+  control,
+  setDidVerify,
+}: {
+  control: Dialog.DialogControlProps
+  setDidVerify: (value: boolean) => void
+}) {
+  const {_} = useLingui()
+  const {currentAccount} = useSession()
+  const agent = useAgent()
+  const t = useTheme()
+  const {openModal} = useModalControls()
+  const {gtMobile} = useBreakpoints()
+
+  const [currentStep, setCurrentStep] = React.useState<
+    'StepOne' | 'StepTwo' | 'StepThree'
+  >('StepOne')
+  const [confirmationCode, setConfirmationCode] = React.useState('')
+  const [isProcessing, setIsProcessing] = React.useState(false)
+  const [error, setError] = React.useState('')
+
+  const uiStrings = {
+    StepOne: {
+      title: _(msg`Verify Your Email`),
+      message: '',
+    },
+    StepTwo: {
+      title: _(msg`Enter Code`),
+      message: _(
+        msg`An email has been sent! Please enter the confirmation code included in the email below.`,
+      ),
+    },
+    StepThree: {
+      title: _(msg`Success!`),
+      message: _(msg`Thank you! Your email has been successfully verified.`),
+    },
+  }
+
+  const onSendEmail = async () => {
+    setError('')
+    setIsProcessing(true)
+    try {
+      await agent.com.atproto.server.requestEmailConfirmation()
+      setCurrentStep('StepTwo')
+    } catch (e: unknown) {
+      setError(cleanError(e))
+    } finally {
+      setIsProcessing(false)
+    }
+  }
+
+  const onVerifyEmail = async () => {
+    setError('')
+    setIsProcessing(true)
+    try {
+      await agent.com.atproto.server.confirmEmail({
+        email: (currentAccount?.email || '').trim(),
+        token: confirmationCode.trim(),
+      })
+    } catch (e: unknown) {
+      setError(cleanError(String(e)))
+      setIsProcessing(false)
+      return
+    }
+
+    setIsProcessing(false)
+    setDidVerify(true)
+    setCurrentStep('StepThree')
+  }
+
+  return (
+    <Dialog.ScrollableInner label={_(msg`Verify email dialog`)}>
```

Comment:
I think on web we can do what we do with prompt and make this _not_ full width. Feels stretched rn.